controllers/helpers.go

```diff
@@ -338,6 +341,64 @@ func GetImage(ctx context.Context, client *prismclientv3.Client, id infrav1.Nuta
 	}
 }
 
+type ImageLookup struct {
+	BaseOS     string
+	K8sVersion string
+}
+
+func GetImageWithFormat(
+	ctx context.Context,
+	client *prismclientv3.Client,
+	imageTemplate,
+	imageLookupBaseOS,
+	k8sVersion *string,
+) (*prismclientv3.ImageIntentResponse, error) {
+	if strings.Contains(*k8sVersion, "v") {
+		k8sVersion = ptr.To(strings.Replace(*k8sVersion, "v", "", 1))
+	}
+	params := ImageLookup{*imageLookupBaseOS, *k8sVersion}
+	t, err := template.New("k8sTemplate").Parse(*imageTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse template given %s %v", *imageTemplate, err)
+	}
+	var templateBytes bytes.Buffer
+	err = t.Execute(&templateBytes, params)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to substitute string %s with params %v error: %w",
+			*imageTemplate,
+			params,
+			err,
+		)
+	}
+	filterString := templateBytes.String()
+	filter := fmt.Sprintf("name==%s$", filterString)
+	responseImages, err := client.V3.ListAllImage(ctx, filter)
```

Comment:
Please do not use FIQL, it contains bugs. We're especially get rid from the FIQL in our projects.

Reply:
We use FIQL in several of our products in NKP and it has been working fine for us. There is an e2e test that exercises this functionality as well and it's proven to work.

Reply:
These situations are entirely different. There will be three different ways to grab an image

1. Name directly
2. UUID directly
3. Image filtering

I could understand that previously we wouldn't want to depend on on filtering, but since there are 2 other options that don't depend on it users can make an informed decision based on their PC version.

Reply:
Situation (2) doesn't require filtering, so that's not part of this conversation.

Situations (1) and (3) both require filtering, and we do client-side filtering in (1). Because we want a consistent implementation, and because FIQL is deprecated, we should therefore use client-side filtering for (2), unless there is a compelling reason to _not_ do so.

Do you see a compelling reason?

(Implementing the client-side filter for (2) seems straightforward.)

Reply:
for the sake of consistency i went with client side filtering for 3.